src/main/java/org/kiwiproject/dropwizard/util/health/MonitoredJobHealthCheck.java

```diff
@@ -0,0 +1,143 @@
+package org.kiwiproject.dropwizard.util.health;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.isNull;
+import static org.kiwiproject.base.KiwiPreconditions.requireNotNull;
+import static org.kiwiproject.base.KiwiStrings.f;
+import static org.kiwiproject.metrics.health.HealthCheckResults.newResultBuilder;
+import static org.kiwiproject.metrics.health.HealthCheckResults.newUnhealthyResult;
+import static org.kiwiproject.metrics.health.HealthCheckResults.newUnhealthyResultBuilder;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.google.common.annotations.VisibleForTesting;
+import io.dropwizard.util.Duration;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.kiwiproject.base.DefaultEnvironment;
+import org.kiwiproject.base.KiwiEnvironment;
+import org.kiwiproject.dropwizard.util.job.MonitoredJob;
+
+import java.time.Instant;
+
+@Slf4j
+@Getter(AccessLevel.PACKAGE) // For testing purposes
+public class MonitoredJobHealthCheck extends HealthCheck {
+
+    @VisibleForTesting
+    static final double DEFAULT_THRESHOLD_FACTOR = 2.0;
+
+    @VisibleForTesting
+    static final Duration MINIMUM_WARNING_THRESHOLD = Duration.minutes(1);
+
+    @VisibleForTesting
+    static final Duration DEFAULT_WARNING_DURATION = Duration.minutes(15);
+
+    private final MonitoredJob job;
```

Comment:
We need to explain what these properties mean. Several are (to me anyway) not entirely clear what they are, e.g. `expectedFrequency`.

Since they are only visible for testing, maybe the explanations should go at the class-level doc, since we cannot doc the builder either. You could use an actual HTML table 😎(see, for example, Dropwizard's `DefaultServerFactory`)